src/providers/ffz/FfzBadges.hpp

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <boost/optional.hpp>
+#include <common/Singleton.hpp>
+
+#include "common/Aliases.hpp"
+
+#include <map>
+#include <vector>
+
+namespace chatterino {
+
+struct Emote;
+using EmotePtr = std::shared_ptr<const Emote>;
+
+class FfzBadges : public Singleton
+{
+public:
+    virtual void initialize(Settings &settings, Paths &paths) override;
+    FfzBadges();
+
+    boost::optional<EmotePtr> getBadge(const UserId &id);
+    boost::optional<QColor> getBadgeColor(const UserId &id);
+
+private:
+    void loadFfzBadges();
+    std::map<QString, int> badgeMap;
+    std::vector<EmotePtr> emotes;
```

Comment:
maybe rename to `badges`? I know the underlying type is an Emote, but in essence what's actually stored is an image used as a badge